torch/_inductor/scheduler.py

```diff
@@ -233,10 +240,6 @@ def allocate(self):
                 )
             )
             and config.inplace_buffers
-            and (
-                not isinstance(V.kernel, torch._inductor.codegen.triton.TritonKernel)
-                or getattr(V.kernel, "mutations", None) is not None
-            )
```

Comment:
This evaluates to `True`  as a `TritonKernel`  always has a `"mutations"` arg and it defaults to an empty set.

Reply:
`TritonTemplateKernel` seems to default `mutations` to `None`. That might be relevant?

Reply:
oops, rolling back